### PR TITLE
fix negative trait bound in outline view (#14044)

### LINKED ISSUE #14044: negative trait bound show incorrectly in vscode outline view
For below codes

```
struct Test;
impl !Unpin for Test {}
```
The outline show `impl Unpin for Test`, without `!`

**rust-analyzer version**: 0.3.1377

**rustc version**: rustc 1.66.0 (69f9c33d7 2022-12-12)

**relevant settings**: default

### PR DESCRIPTION
**Pull request by gftea**
try to fix and close #14044

### CODE REVIEW

File: crates/ide/src/file_structure.rs [hunk 1]
```diff
@@ -160,7 +160,10 @@ fn structure_node(node: &SyntaxNode) -> Option<StructureNode> {
                 let label = match target_trait {
                     None => format!("impl {}", target_type.syntax().text()),
                     Some(t) => {
-                        format!("impl {} for {}", t.syntax().text(), target_type.syntax().text(),)
+                        format!("impl {}{} for {}", 
+                        it.excl_token().map(|x| x.to_string()).unwrap_or_default(), 
+                        t.syntax().text(), 
+                        target_type.syntax().text(),)
```
Comment (jonas-schievink) [inline crates/ide/src/file_structure.rs:166]: The change looks correct, but please format the code to be a bit more readable:

```suggestion
                        format!("impl {}{} for {}", 
                            it.excl_token().map(|x| x.to_string()).unwrap_or_default(), 
                            t.syntax().text(), 
                            target_type.syntax().text(),
                        )
```

**Comment by gftea** [comment]
Hi,

I think it is good to improve the contribution guide saying that rustfmt should be used to format the codes

**Comment by gftea** [comment]
@jonas-schievink is it OK to merge now?

**Comment by gftea** [comment]
Hi, how to track when it is released to new vscode extension?

**Comment by Veykril** [comment]
Whoops wrong command
@bors r+

It will release next monday for stable, and in roughly 5 hours on the nightly release

**Pull request merged**